### PR TITLE
clean up discreteuniformgenerator usage and eliminate unnecessary use of SetFormat

### PR DESCRIPTION
**Pull request by pascoec**

### CODE REVIEW

File: src/binfhe/lib/rgsw-acc-cggi.cpp [hunk 1]
```diff
@@ -75,7 +75,6 @@ RingGSWEvalKey RingGSWAccumulatorCGGI::KeyGenCGGI(const std::shared_ptr<RingGSWC
 
     DiscreteUniformGeneratorImpl<NativeVector> dug;
     NativeInteger Q{params->GetQ()};
-    dug.SetModulus(Q);
```
Comment (dsuponitskiy) [inline src/binfhe/lib/rgsw-acc-cggi.cpp:78]: would like to make sure that dug doesn't need to be constructed with the modulus "Q".

File: src/binfhe/lib/rgsw-acc-dm.cpp [hunk 2]
```diff
@@ -85,7 +85,6 @@ RingGSWEvalKey RingGSWAccumulatorDM::KeyGenDM(const std::shared_ptr<RingGSWCrypt
 
     DiscreteUniformGeneratorImpl<NativeVector> dug;
     NativeInteger Q{params->GetQ()};
-    dug.SetModulus(Q);
```
Comment (dsuponitskiy) [inline src/binfhe/lib/rgsw-acc-dm.cpp:88]: would like to make sure that dug doesn't need to be constructed with the modulus "Q".

File: src/binfhe/lib/rgsw-acc-lmkcdey.cpp [hunk 3]
```diff
@@ -167,7 +167,6 @@ RingGSWEvalKey RingGSWAccumulatorLMKCDEY::KeyGenLMKCDEY(const std::shared_ptr<Ri
 
     DiscreteUniformGeneratorImpl<NativeVector> dug;
     NativeInteger Q{params->GetQ()};
-    dug.SetModulus(Q);
```
Comment (dsuponitskiy) [inline src/binfhe/lib/rgsw-acc-lmkcdey.cpp:170]: would like to make sure that dug doesn't need to be constructed with the modulus "Q".

File: src/binfhe/lib/rgsw-acc-lmkcdey.cpp [hunk 4]
```diff
@@ -209,7 +208,6 @@ RingGSWEvalKey RingGSWAccumulatorLMKCDEY::KeyGenAuto(const std::shared_ptr<RingG
...
```
Comment (dsuponitskiy) [inline src/binfhe/lib/rgsw-acc-lmkcdey.cpp:212]: would like to make sure that dug doesn't need to be constructed with the modulus "Q".

File: src/core/extras/ntt2.cpp [hunk 5]
```diff
@@ -134,10 +134,8 @@ void test_NTT(const usint level, const usint nloop) {
     ILParams params1(m, q1, rootOfUnity1);
     auto x1p = std::make_shared<ILParams>(params1);
 
-    Poly::DugType dug1;
-    dug1.SetModulus(q1);  // random # generator to use
-
     // two vectors
+    Poly::DugType dug1;
```
Comment (dsuponitskiy) [inline src/core/extras/ntt2.cpp:138]: would like to make sure that dug1 doesn't need to be constructed with the modulus "q1".

File: src/core/extras/ntt2.cpp [hunk 6]
```diff
@@ -166,8 +164,6 @@ void test_NTT(const usint level, const usint nloop) {
     auto x2p = std::make_shared<ILParams>(params2);
 
     Poly::DugType dug2;
-    dug2.SetModulus(q1);  // random # generator to use
```
Comment (dsuponitskiy) [inline src/core/extras/ntt2.cpp:169]: would like to make sure that dug2 doesn't need to be constructed with the modulus "q1".

File: src/core/extras/ntt2.cpp [hunk 7]
```diff
@@ -196,10 +192,8 @@ void test_NTT(const usint level, const usint nloop) {
     ILParams params3(m, q3, rootOfUnity3);
     auto x3p = std::make_shared<ILParams>(params3);
 
-    Poly::DugType dug3;
-    dug3.SetModulus(q1);  // random # generator to use
```
Comment (dsuponitskiy) [inline src/core/extras/ntt2.cpp:200]: would like to make sure that dug3 doesn't need to be constructed with the modulus "q1".

File: src/core/unittest/UnitTestTrapdoor.cpp [hunk 8]
```diff
@@ -273,8 +273,7 @@ TEST(UTTrapdoor, TrapDoorGaussGqSampTest) {
     double sigma  = (base + 1) * SIGMA;
 
     Poly::DggType dgg(sigma);
-    Poly::DugType dug = Poly::DugType();
-    dug.SetModulus(modulus);
+    Poly::DugType dug;
```
Comment (dsuponitskiy) [inline src/core/unittest/UnitTestTrapdoor.cpp:276]: constructed without "modulus"

File: src/core/unittest/UnitTestPolyElements.cpp [hunk 9]
```diff
@@ -592,8 +591,7 @@ void Poly_mod_ops_on_two_elements(const std::string& msg) {
 
     auto ilparams = std::make_shared<ParmType>(order, primeModulus, primitiveRootOfUnity);
 
-    typename Element::DugType distrUniGen = typename Element::DugType();
-    distrUniGen.SetModulus(primeModulus);
+    typename Element::DugType distrUniGen;
```
Comment (dsuponitskiy) [inline src/core/unittest/UnitTestPolyElements.cpp:594]: constructed without "primeModulus"

File: src/core/unittest/UnitTestTrapdoor.cpp [hunk 10]
```diff
@@ -396,8 +395,7 @@ TEST(UTTrapdoor, TrapDoorGaussGqSampTestBase1024) {
     double sigma  = (base + 1) * SIGMA;
 
     Poly::DggType dgg(SIGMA);
-    Poly::DugType dug = Poly::DugType();
-    dug.SetModulus(modulus);
+    Poly::DugType dug;
```
Comment (dsuponitskiy) [inline src/core/unittest/UnitTestTrapdoor.cpp:398]: constructed without "modulus"

File: src/core/unittest/UnitTestTrapdoor.cpp [hunk 11]
```diff
@@ -491,8 +489,7 @@ TEST(UTTrapdoor, TrapDoorGaussSampTest) {
     // Format::EVALUATION);
 
     Poly::DggType dgg(sigma);
-    Poly::DugType dug = Poly::DugType();
-    dug.SetModulus(modulus);
+    Poly::DugType dug;
```
Comment (dsuponitskiy) [inline src/core/unittest/UnitTestTrapdoor.cpp:492]: constructed without "modulus"

File: src/binfhe/lib/rgsw-acc-cggi.cpp (hunk 1 above)
Comment (pascoec) [reply src/binfhe/lib/rgsw-acc-cggi.cpp:78]: It Doesn't, which is why I removed it. Modulus is set on line 82 when dug is first used.

File: src/binfhe/lib/rgsw-acc-dm.cpp (hunk 2 above)
Comment (pascoec) [reply src/binfhe/lib/rgsw-acc-dm.cpp:88]: same

File: src/core/unittest/UnitTestTrapdoor.cpp (hunk 11 above)
Comment (pascoec) [reply src/core/unittest/UnitTestTrapdoor.cpp:492]: same

File: src/core/unittest/UnitTestTrapdoor.cpp (hunk 10 above)
Comment (pascoec) [reply src/core/unittest/UnitTestTrapdoor.cpp:398]: same

File: src/core/unittest/UnitTestPolyElements.cpp (hunk 9 above)
Comment (pascoec) [reply src/core/unittest/UnitTestPolyElements.cpp:594]: same

File: src/core/unittest/UnitTestTrapdoor.cpp (hunk 8 above)
Comment (pascoec) [reply src/core/unittest/UnitTestTrapdoor.cpp:276]: same

File: src/core/extras/ntt2.cpp (hunk 7 above)
Comment (pascoec) [reply src/core/extras/ntt2.cpp:200]: same

File: src/core/extras/ntt2.cpp (hunk 6 above)
Comment (pascoec) [reply src/core/extras/ntt2.cpp:169]: same

File: src/core/extras/ntt2.cpp (hunk 5 above)
Comment (pascoec) [reply src/core/extras/ntt2.cpp:138]: same

**Review by dsuponitskiy — APPROVED** [review]
Looks good

**Pull request merged**